common/beerocks/tlvf/src/include/beerocks/tlvf/beerocks_message.h

```diff
@@ -61,30 +61,48 @@ class message_com {
         return std::dynamic_pointer_cast<beerocks_message::cACTION_HEADER>(cmdu.getClass(1));
     }
 
-    template <class T>
-    static std::shared_ptr<T>
-    add_intel_vs_data(ieee1905_1::CmduMessageTx &cmdu_tx,
-                      std::shared_ptr<ieee1905_1::tlvVendorSpecific> vs_tlv, uint16_t id = 0)
+    static std::shared_ptr<beerocks_header> get_beerocks_header(ieee1905_1::CmduMessage &cmdu)
     {
-        beerocks_message::eAction action;
-        std::shared_ptr<T> p_class = nullptr;
-
-        auto actionhdr = cmdu_tx.addClass<beerocks_message::cACTION_HEADER>();
-        if (!actionhdr) {
-            std::cout << "beerocks_message.h[ " << __LINE__ << "]: " << __FUNCTION__ << " failed!"
-                      << std::endl;
+        auto tlv = cmdu.msg.getClass<ieee1905_1::tlvVendorSpecific>();
+        if (!tlv)
             return nullptr;
-        }
-        vs_tlv->length() += beerocks_message::cACTION_HEADER::get_initial_size();
+        if (!intel_oui(tlv))
+            return nullptr;
+        return std::dynamic_pointer_cast<beerocks_header>(tlv->getInnerClassList());
+    }
 
-        p_class = cmdu_tx.addClass<T>();
-        if (!p_class) {
+    template <class T>
+    static std::shared_ptr<beerocks_header>
+    add_intel_vs_data(std::shared_ptr<ieee1905_1::tlvVendorSpecific> tlv,
+                                  uint16_t id = 0)
+    {
+        // Allocate maximum allowed length for the payload, so it can accommodate variable length
+        // data inside the internal TLV list.
+        // On finalize(), the buffer is shrunk back to its real size.
+        size_t payload_length;
+        const size_t max_vs_tlv_length = ieee1905_1::CmduMessage::kMaxCmduLength -
+                                         ieee1905_1::CmduMessage::kCmduHeaderLength -
+                                         ieee1905_1::tlvVendorSpecific::get_initial_size() -
+                                         ieee1905_1::tlvEndOfMessage::get_initial_size();
```

Comment:
I still don't understand why we don't just allocate the rest of the buffer here. That's basically what we've done up to now and it seems to work...

Reply:
Didn't cherry pick the commit yet, we will